### PR TITLE
Remove ListRepos in favor of internal/db/RepoStore#List

### PR DESCRIPTION
**Pull request by asdine**
This PR removes `ListRepos` from the repo-updater store in favor of [internal/db/RepoStore#List](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/internal/db/repos.go?utm_source=VSCode-1.1.0#L563:23).

### CODE REVIEW

File: cmd/repo-updater/repoupdater/server_test.go [hunk 1]
```diff
@@ -1404,20 +1410,25 @@ func formatJSON(s string) string {
 type storeWithErrors struct {
```
Comment (ryanslade) [inline cmd/repo-updater/repoupdater/server_test.go:1410]: Looks like this can be removed

File: enterprise/cmd/repo-updater/authz/perms_syncer.go [hunk 2]
```diff
@@ -284,9 +284,9 @@ func (s *PermsSyncer) syncUserPerms(ctx context.Context, userID int32, noPerms b
 	var rs []*types.Repo
 	if len(repoSpecs) > 0 {
 		// Get corresponding internal database IDs
-		rs, err = s.reposStore.ListRepos(ctx, repos.StoreListReposArgs{
+		rs, err = s.reposStore.List(ctx, db.ReposListOptions{
 			ExternalRepos: repoSpecs,
-			PrivateOnly:   true,
+			OnlyPrivate:   true,
```
Comment (ryanslade) [inline enterprise/cmd/repo-updater/authz/perms_syncer.go:289]: Out of interest, why was this renamed?

File: enterprise/cmd/repo-updater/authz/perms_syncer.go [hunk 3]
```diff
@@ -449,6 +449,8 @@ func (s *PermsSyncer) syncRepoPerms(ctx context.Context, repoID api.RepoID, noPe
 		return errors.Wrap(err, "set repository pending permissions")
 	}
 
+	fmt.Println(8)
```
Comment (ryanslade) [inline enterprise/cmd/repo-updater/authz/perms_syncer.go:452]: Debugging?

File: internal/repos/store.go [hunk 4]
```diff
@@ -356,145 +330,6 @@ WHERE deleted_at IS NULL
...
-			sqlf.Sprintf("(%s)", predQ),
-			joinFilter,
-			limit,
-		)
-	}, nil
-}
-
 func (s *DBStore) ListExternalRepoSpecs(ctx context.Context) (map[api.ExternalRepoSpec]struct{}, error) {
 	const ListExternalRepoSpecsQueryFmtstr = `
 -- source: internal/repos/store.go:DBStore.ListExternalRepoSpecs
```
Comment (ryanslade) [inline internal/repos/store.go:335]: Not able to place a comment at the right place, but see the `golang-ci` checks below, there's some dead code that can be removed.

**Review by ryanslade — APPROVED** [review]

File: enterprise/cmd/repo-updater/authz/perms_syncer.go (hunk 2 above)
Comment (asdine) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go:289]: This is the name of the option in internal/db, it was already named like this. Naming conventions between the two stores are very different.

File: enterprise/cmd/repo-updater/authz/perms_syncer.go (hunk 3 above)
Comment (asdine) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go:452]: Haha good catch
